domain/network/state/space_test.go

```diff
@@ -33,8 +33,8 @@ func (s *stateSuite) TestAddSpace(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	err = st.AddSubnet(
 		ctx.Background(),
+		subnetUUID.String(),
 		network.SubnetInfo{
-			ID:                network.Id(subnetUUID.String()),
```

Comment:
I actually like the idea of upstream caller being able to supply this ID, as that's what I want to happen eventually.

We can land this as is, but I'd like a change for this and other services to:
- Just work if the incoming info has a UUID.
- Populate it with a generated one if empty.

The caller always gets the UUID inserted, so this is safe.

Reply:
Yeah indeed I think it's a good convention so I did it for this service (only AddSubnet has this type of input). See 8c41e31aac